ballerina/utils.bal

```diff
@@ -15,21 +15,154 @@
+    "0", "1", "2", "3", "4", "5", 
+    "6", "7", "8", "9", "a", "b", 
+    "c", "d", "e", "f", "g", "h", 
+    "i", "j", "k", "l", "m", "n", 
+    "o", "p", "q", "r", "s", "t", 
+    "u", "v", "w", "x", "y", "z"
+];
 
 isolated function getRandomUUID() returns handle = @java:Method {
     name: "randomUUID",
     'class: "java.util.UUID"
 } external;
 
-isolated function getBytesFromUUID(string uuid) returns byte[] = @java:Method {
-    name: "getBytesFromUUID",
-    'class: "io.ballerina.stdlib.uuid.nativeimpl.Util"
-} external;
+isolated function getBytesFromUUID(string uuid) returns byte[]|Error {
+    int msb = check getMostSignificantBits(uuid);
+    int lsb = check getLeastSignificantBits(uuid);
+    return getBytesFromSignificantBits(msb, lsb);
+}
 
-isolated function getUUIDFromBytes(byte[] uuid) returns string = @java:Method {
-    name: "getUUIDFromBytes",
-    'class: "io.ballerina.stdlib.uuid.nativeimpl.Util"
-} external;
+isolated function getUUIDFromBytes(byte[] uuid) returns string {
+    int msb = ((uuid[0] & 0xFF) << 56) |
+            ((uuid[1] & 0xFF) << 48) |
+            ((uuid[2] & 0xFF) << 40) |
+            ((uuid[3] & 0xFF) << 32) |
+            ((uuid[4] & 0xFF) << 24) |
+            ((uuid[5] & 0xFF) << 16) |
+            ((uuid[6] & 0xFF) << 8) |
+            ((uuid[7] & 0xFF) << 0);
+
+    int lsb = ((uuid[8] & 0xFF) << 56) |
+            ((uuid[9] & 0xFF) << 48) |
+            ((uuid[10] & 0xFF) << 40) |
+            ((uuid[11] & 0xFF) << 32) |
+            ((uuid[12] & 0xFF) << 24) |
+            ((uuid[13] & 0xFF) << 16) |
+            ((uuid[14] & 0xFF) << 8) |
+            ((uuid[15] & 0xFF) << 0);
+
+    return getUuidFromSignificantBits(msb, lsb);
+}
+
+isolated function getMostSignificantBits(string uuid) returns int|Error {
+    Uuid u = check toRecord(uuid);
+
+    int mostSigBits = u.timeLow & 0xffffffff;
+    mostSigBits <<= 16;
+    mostSigBits |= u.timeMid & 0xffff;
+    mostSigBits <<= 16;
+    mostSigBits |= u.timeHiAndVersion & 0xffff;
+
+    return mostSigBits;
+}
+
+isolated function getLeastSignificantBits(string uuid) returns int|Error {
+    Uuid u = check toRecord(uuid);
+
+    int leastSigBits;
+    string clockSeq = regex:split(uuid, "-")[3];
+    int|error clockSeqInt = ints:fromHexString(clockSeq);
+    if clockSeqInt is int {
+        leastSigBits = clockSeqInt & 0xffff;
+    } else {
+        return error Error("Failed to get clock sequence value of the uuid");
+    }
+
+    leastSigBits <<= 48;
+    leastSigBits |= u.node & 0xffffffffffff;
+
+    return leastSigBits;
+}
+
+isolated function get64LeastSignificantBitsForVersion1() returns int {
+    int random63BitLong = nextLong() & 0x3FFFFFFFFFFFFFFF;
+    int variant3BitFlag = -0x8000000000000000;
+    return random63BitLong + variant3BitFlag;
+}
+
+isolated function get64MostSignificantBitsForVersion1() returns int {
+    // The gregorian time(1582-10-15 0:0:0) is a constant
+    int gregorianTimeInSeconds = 12219292800;
```

Comment:
shall we move this to a const

Reply:
seems like `gregorianTimeInSeconds` is decimal. shall we define it as a decimal number and avoid casting

Reply:
Updated in [036e487](https://github.com/ballerina-platform/module-ballerina-uuid/pull/376/commits/036e4872243b18edd5c1f692a4bb06df0c995232)